server/backends/blobstore/gcs/gcs.go

```diff
@@ -312,10 +312,19 @@ func (d *decompressingCloser) Close() error {
 	return firstError
 }
 
-func (g *GCSBlobStore) Reader(ctx context.Context, blobName string) (io.ReadCloser, error) {
+func (g *GCSBlobStore) Reader(ctx context.Context, blobName string, offset, limit int64) (io.ReadCloser, error) {
 	ctx, spn := tracing.StartSpan(ctx)
 	defer spn.End()
-	reader, err := g.bucketHandle.Object(blobName).NewReader(ctx)
+	if offset < 0 {
+		// GCS treats negative offsets as relative to the end of the object, but
+		// we don't generally support that.
+		offset = 0
+	}
+	if limit == 0 {
+		// We use 0 to request the whole object, but GCS uses -1.
+		limit = -1
+	}
+	reader, err := g.bucketHandle.Object(blobName).NewRangeReader(ctx, offset, limit)
```

Comment:
are NewRangeReader(ctx, 0, -1) and NewReader(ctx) functionally the same? just double checking this won't impact performance somehow